lib/redis/index.ts

```diff
@@ -201,12 +202,18 @@ Redis.prototype.parseOptions = function() {
       defaults(this.options, arg);
     } else if (typeof arg === "string") {
       defaults(this.options, parseURL(arg));
+      if (arg.startsWith("rediss://")) {
+        isTls = true;
+      }
     } else if (typeof arg === "number") {
       this.options.port = arg;
     } else {
       throw new Error("Invalid argument " + arg);
     }
   }
+  if (isTls) {
+    defaults(this.options, { tls: { servername: this.options.host } });
```

Comment:
I'm not sure if setting `tls` to `this.options.host` instead of `true` applies to more scenarios, but I think at least we need to check whether `this.options.host` is a hostname instead of an IP address.

As for me, I prefer the way that we just default `tls` options to `true` without setting any other properties (like `servername`) and let users handle it. The improvement we should make here is to keep the options provided by users not be overwritten.

Reply:
> I'm not sure if setting `tls` to `this.options.host` instead of `true` applies to more scenarios, but I think at least we need to check whether `this.options.host` is a hostname instead of an IP address.

I'm happy to make that modification if you'd like.

Setting `tls.options.host` to the hostname of the server is necessary to use many shared Redis hosting services, including Compose.io and Redis Labs. It reduces boilerplate and seems like a reasonable default as this is not really a fringe use case, but a common scenario for TLS.

> The improvement we should make here is to keep the options provided by users not be overwritten.

Alternatively, #949 does just that!